### PR TITLE
Update hello samples

### PR DESCRIPTION
**Pull request by tsurdilo**
Signed-off-by: Tihomir Surdilovic <tihomir@temporal.io>

## What was changed:

1) Added comments and descriptions to all of the hello examples
2) Updates examples to all include a workflow id
3) made some small updates to examples

## Why?

When I was first learning about Temporal these examples were the first thing I looked at. 
I wanted to add these changes to make it easier for people that learn from them as well
to be able to easier understand the code.

## Checklist

1. Closes issue: 

2. How was this tested:

3. Any docs updates needed?

### CODE REVIEW

File: src/main/java/io/temporal/samples/hello/HelloActivity.java [hunk 1]
```diff
@@ -74,39 +104,80 @@ public String getGreeting(String name) {
...
 
-    // worker factory that can be used to create workers for specific task queues
+    /*
+     * Define the workflow factory. It is used to create workflow workers for a specific task queue.
+     */
     WorkerFactory factory = WorkerFactory.newInstance(client);
-    // Worker that listens on a task queue and hosts both workflow and activity implementations.
+
+    /*
+     * Define the workflow worker. Workflow workers listen to a defined task queue and process
```
Comment (vitarb) [inline src/main/java/io/temporal/samples/hello/HelloActivity.java:142]: s/listen/listens

File: src/main/java/io/temporal/samples/hello/HelloActivity.java [hunk 2]
```diff
@@ -33,34 +33,64 @@
...
+ * https://github.com/temporalio/temporal/blob/master/README.md#download-and-start-temporal-server-locally
  */
 public class HelloActivity {
 
-  static final String TASK_QUEUE = "HelloActivity";
+  // Define the task queue name
+  static final String TASK_QUEUE = "HelloActivityTaskQueue";
+
+  // Define our workflow unique id
+  static final String WORKFLOW_ID = "HelloActivityWorkflow";
```
Comment (vitarb) [inline src/main/java/io/temporal/samples/hello/HelloActivity.java:48]: Should we mention about workflow id reuse policy here?

File: src/main/java/io/temporal/samples/hello/HelloActivity.java [hunk 3]
```diff
@@ -33,34 +33,64 @@
...
 
-  /** Workflow interface has to have at least one method annotated with @WorkflowMethod. */
+  /**
+   * Define the Workflow Interface. It must contain at least one method annotated
```
Comment (vitarb) [inline src/main/java/io/temporal/samples/hello/HelloActivity.java:51]: I believe it must have one and only one workflow method. Please double check, but I don't think we allow having multiple.

File: src/main/java/io/temporal/samples/hello/HelloActivity.java [hunk 4]
```diff
@@ -33,34 +33,64 @@
...
+   * with @WorkflowMethod
+   *
+   * @see io.temporal.workflow.WorkflowInterface
+   * @see io.temporal.workflow.WorkflowMethod
+   */
   @WorkflowInterface
   public interface GreetingWorkflow {
+
+    /**
+     * Define the workflow method. This method is executed when the workflow is started. The
```
Comment (vitarb) [inline src/main/java/io/temporal/samples/hello/HelloActivity.java:61]: s/Define the workflow method.// - this sentence carries no meaning, I think we should just remove it.

File: src/main/java/io/temporal/samples/hello/HelloActivity.java [hunk 5]
```diff
@@ -33,34 +33,64 @@
...
+   * Annotating activity methods with @ActivityMethod is optional
+   *
+   * @see io.temporal.activity.ActivityInterface
+   * @see io.temporal.activity.ActivityMethod
+   */
   @ActivityInterface
   public interface GreetingActivities {
+
+    // Define your activity method which can be called during workflow execution
     @ActivityMethod
```
Comment (vitarb) [inline src/main/java/io/temporal/samples/hello/HelloActivity.java:79]: This annotation is not required, I think we should just remove it.

File: src/main/java/io/temporal/samples/hello/HelloActivity.java [hunk 6]
```diff
@@ -33,34 +33,64 @@
...
     String composeGreeting(String greeting, String name);
   }
 
-  /** GreetingWorkflow implementation that calls GreetingsActivities#composeGreeting. */
+  // Define the workflow implementation. It implements our getGreeting workflow method
```
Comment (vitarb) [inline src/main/java/io/temporal/samples/hello/HelloActivity.java:83]: s/.It implements our/which implements

File: src/main/java/io/temporal/samples/hello/HelloActivity.java [hunk 7]
```diff
@@ -33,34 +33,64 @@
...
+     * workflow completes when the workflow method finishes execution.
+     */
     @WorkflowMethod
     String getGreeting(String name);
   }
 
-  /** Activity interface is just a POJI. */
+  /**
+   * Define the Activity Interface. Workflow methods can call activities during execution.
+   * Annotating activity methods with @ActivityMethod is optional
```
Comment (vitarb) [inline src/main/java/io/temporal/samples/hello/HelloActivity.java:70]: As mentioned below, I think we should just remove it from this sample.

File: src/main/java/io/temporal/samples/hello/HelloActivity.java [hunk 8]
```diff
@@ -33,34 +33,64 @@
...
-  /** GreetingWorkflow implementation that calls GreetingsActivities#composeGreeting. */
+  // Define the workflow implementation. It implements our getGreeting workflow method
   public static class GreetingWorkflowImpl implements GreetingWorkflow {
 
     /**
-     * Activity stub implements activity interface and proxies calls to it to Temporal activity
-     * invocations. Because activities are reentrant, only a single stub can be used for multiple
-     * activity invocations.
+     * Define the GreetingActivities stub. Activity stubs implements activity interfaces and proxy
+     * calls to it to Temporal activity invocations. Since Temporal activities are reentrant, a
```
Comment (vitarb) [inline src/main/java/io/temporal/samples/hello/HelloActivity.java:88]: I would replace "Activity stubs implements activity interfaces and proxy calls to it to Temporal activity invocations." with "Activity stubs are proxies for activity invocations that are executed outside of the workflow thread on the activity worker, that can be on a different host. Temporal is going to dispatch activity results back to the workflow and unblock the stub as soon as activity is completed on the activity worker."

File: src/main/java/io/temporal/samples/hello/HelloActivity.java [hunk 9]
```diff
@@ -74,39 +104,80 @@ public String getGreeting(String name) {
     }
   }
 
+  /**
+   * Implementation of our workflow activity interface. It overwrites our defined composeGreeting
```
Comment (vitarb) [inline src/main/java/io/temporal/samples/hello/HelloActivity.java:108]: I would rephrase:
"Simple activity implementation, that concatenates two strings."

File: src/main/java/io/temporal/samples/hello/HelloActivity.java [hunk 10]
```diff
@@ -74,39 +104,80 @@ public String getGreeting(String name) {
...
   static class GreetingActivitiesImpl implements GreetingActivities {
     @Override
     public String composeGreeting(String greeting, String name) {
       return greeting + " " + name + "!";
     }
   }
 
+  /**
+   * With our Workflow and Activities defined, we can now start execution. The main method is our
+   * workflow starter.
```
Comment (vitarb) [inline src/main/java/io/temporal/samples/hello/HelloActivity.java:120]: We should say that it starts the worker and then the workflow.

File: src/main/java/io/temporal/samples/hello/HelloActivity.java [hunk 11]
```diff
@@ -74,39 +104,80 @@ public String getGreeting(String name) {
...
+   */
   public static void main(String[] args) {
-    // gRPC stubs wrapper that talks to the local docker instance of temporal service.
+
+    /*
+     * Define the workflow service. It is a gRPC stubs wrapper which talks to the docker instance of
```
Comment (vitarb) [inline src/main/java/io/temporal/samples/hello/HelloActivity.java:125]: I would remove word docker from here, we should be able to run samples against our cloud service in the future.

File: src/main/java/io/temporal/samples/hello/HelloActivity.java [hunk 12]
```diff
@@ -74,39 +104,80 @@ public String getGreeting(String name) {
...
+
+    /*
+     * Define the workflow worker. Workflow workers listen to a defined task queue and process
+     * workflows and activities.
+     */
     Worker worker = factory.newWorker(TASK_QUEUE);
-    // Workflows are stateful. So you need a type to create instances.
+
+    /*
+     * Register our workflow implementation with the worker. Since workflows are stateful in nature,
```
Comment (vitarb) [inline src/main/java/io/temporal/samples/hello/HelloActivity.java:148]: 2nd sentence doesn't make much sense to me. Should we just say that workflow implementations must be known to the worker at the runtime in order to dispatch workflow tasks?

File: src/main/java/io/temporal/samples/hello/HelloActivity.java [hunk 13]
```diff
@@ -33,34 +33,64 @@
...
+     * Define the workflow method. This method is executed when the workflow is started. The
+     * workflow completes when the workflow method finishes execution.
+     */
     @WorkflowMethod
     String getGreeting(String name);
   }
 
-  /** Activity interface is just a POJI. */
+  /**
+   * Define the Activity Interface. Workflow methods can call activities during execution.
```
Comment (vitarb) [inline src/main/java/io/temporal/samples/hello/HelloActivity.java:69]: I would expand this a little bit.
"Activities are building blocks of any temporal workflow, and contain any business logic, that could perform long running computation, network calls, etc."

File: src/main/java/io/temporal/samples/hello/HelloActivity.java (hunk 3 above)
Comment (vitarb) [inline src/main/java/io/temporal/samples/hello/HelloActivity.java:51]: I would also say here that workflow code is core processing logic that shouldn't contain any heavyweight computation, non deterministic code, network calls, database operations, etc. All those things should live in activities.

File: src/main/java/io/temporal/samples/hello/HelloActivity.java [hunk 14]
```diff
@@ -74,39 +104,80 @@ public String getGreeting(String name) {
...
-    // Activities are stateless and thread safe. So a shared instance is used.
+
+    /*
+     Register our workflow activity implementation with the worker. Since workflow activities are
+     stateless and thread-safe, we need to register a shared instance.
+    */
     worker.registerActivitiesImplementations(new GreetingActivitiesImpl());
-    // Start listening to the workflow and activity task queues.
+
+    // Start all the workers registered for a specific task queue.
```
Comment (vitarb) [inline src/main/java/io/temporal/samples/hello/HelloActivity.java:159]: I would also say what happens here. Like start workers, which would start polling for workflow and activity tasks.

File: src/main/java/io/temporal/samples/hello/HelloActivity.java [hunk 15]
```diff
@@ -74,39 +104,80 @@ public String getGreeting(String name) {
...
     factory.start();
 
-    // Start a workflow execution. Usually this is done from another program.
-    // Uses task queue from the GreetingWorkflow @WorkflowMethod annotation.
+    // Create our workflow client stub. It is used to start our workflow execution.
```
Comment (vitarb) [inline src/main/java/io/temporal/samples/hello/HelloActivity.java:162]: nit. Here and in other places, I would remove word "our".

File: src/main/java/io/temporal/samples/hello/HelloActivity.java (hunk 1 above)
Comment (tsurdilo) [reply src/main/java/io/temporal/samples/hello/HelloActivity.java:142]: i think with "workflow workers listen" is correct (workers plural).  could change it to "A workflow worker listens" if you think that is better. let me know

File: src/main/java/io/temporal/samples/hello/HelloActivity.java (hunk 2 above)
Comment (tsurdilo) [reply src/main/java/io/temporal/samples/hello/HelloActivity.java:48]: i added a workflow id to each example so that in the web ui it does not display the unique generated one. when running multiple examples locally it helps with figuring out which workflow run is associated with what example.
can you show me maybe what this workflow reuse policy is so i can add ? you mean on like on continue run or something else ?

File: src/main/java/io/temporal/samples/hello/HelloActivity.java (hunk 3 above)
Comment (tsurdilo) [reply src/main/java/io/temporal/samples/hello/HelloActivity.java:51]: +1 yes double checked only one WorkflowMethod annotated is allowed. Will fix everywhere

**Comment by tsurdilo** [comment]
@vitarb thank you for the review! updated per your comments. fixed everything except first two.

File: src/main/java/io/temporal/samples/hello/HelloActivity.java (hunk 1 above)
Comment (Sushisource) [reply src/main/java/io/temporal/samples/hello/HelloActivity.java:140]: Agree it's correct as is

**Review by vitarb — APPROVED** [review]

File: src/main/java/io/temporal/samples/hello/HelloActivity.java (hunk 2 above)
Comment (vitarb) [reply src/main/java/io/temporal/samples/hello/HelloActivity.java:47]: See https://docs.temporal.io/docs/concept-workflows/#workflow-id
